client/state/data-layer/wpcom/posts/revisions/index.js

```diff
@@ -64,8 +74,17 @@ export const receiveError = ( { dispatch }, { siteId, postId }, next, rawError )
  * @param {Array} revisions raw data from post revisions API
  */
 export const receiveSuccess = ( { dispatch }, { siteId, postId }, next, revisions ) => {
+	const normalizedRevisions = map( revisions, normalizeRevision );
+
+	forEach( normalizedRevisions, ( revision, index ) => {
+		revision.changes = index === normalizedRevisions.length - 1
+			? { added: 0, removed: 0 }
+			: countDiffWords( diffWords( normalizedRevisions[ index + 1 ].content, revision.content ) );
```

Comment:
I went this way precisely because it's quite expansive [0]: persisting the summary results along the revisions speeds up the first draw of the revisions list after a reload compared to the equivalent memoized-selector-based solution. That said, it's also partially un-normalizing the revision data in Redux as you get redundant data (the summary can be recomputed from the contents).

> Maybe not needed but what if we want to diff an old revision with the last one for example.

The ability to diff between any 2 revisions is dropped for now: https://calypsop2.wordpress.com/2017/04/19/editor-revisions-i3/#comment-7039
That said, for those features outside the "main" one, it's quite easy to recompute the diff between two revisions with the utils functions introduced here.

As an interesting footnote here: we can most likely go even further and only update the Redux state if we receive deleted or new revisions if we make the assumption that revisions are immutable, AFAICS, there is no way of editing a revision from the API or in wp-admin.

[0] Myers algorithm (used by jsdiff) has a pretty interesting time complexity for revisions like object: it's in O(ND) with N the sum of the number of words in revisions A and B, and D the size of the shortest edit sequence A -> B. If we suppose that most revisions are incremental updates on a post, D should stay small :)